File: CCC '07/Junior/J5 - Keep on Truckin'.py
# https://dmoj.ca/problem/ccc07j5
# CCC '07 J5 - Keep on Truckin'

motel = [0, 990, 1010, 1970, 2030, 2940, 3060, 3930, 4060, 4970, 5030, 5990, 6010, 7000]

a = int(input())
b = int(input())
m = int(input())

for i in range(m):
  mi = int(input())
  motel.append(mi)
motel.sort()

combs = [0 for i in range(len(motel))]
combs[0] = 1

for i in range(1,len(motel)):
  for j in range(i-1, -1, -1):
    if motel[i] - motel[j] >= a and motel[i] - motel[j] <= b:
      combs[i]+=combs[j]
print(combs[-1])

File: Dmoj/CCC '07/Junior/J5 - Keep on Truckin'.cpp
// https://dmoj.ca/problem/ccc07j5
// CCC '07 J5 - Keep on Truckin'

#include <iostream>
#include <vector>
using namespace std;

vector<long long> motels = {0, 990, 1010, 1970, 2030, 2940, 3060, 3930, 4060, 4970, 5030, 5990, 6010, 7000};
int a, b, m; 
int main() {
  cin >> a >> b >> m;
  for (long long i = 0; i<m; i++) {
    long long mi; cin >> mi;
    motels.push_back(mi);
  }
  sort(motels.begin(), motels.end());
  long long comb[motels.size()];
  for (long long i = 0; i< motels.size(); i++) {
    comb[i] = 0;
  }
  comb[0] = 1;
  for (long long i = 1; i<motels.size(); i++) {
    for (long long j = i-1; j>-1; j--) {
      if (motels[i] - motels[j] >= a && motels[i] - motels[j] <= b) {
        comb[i]+=comb[j];
      }
    }
  }
  cout << comb[motels.size()-1] << endl;
  return 0;
}